src/typescript/frontend/src/lib/aptos-indexer/fungible-assets.ts

```diff
@@ -0,0 +1,115 @@
+// cspell:word ilike
+
+import { encodeEmojis, SYMBOL_EMOJIS } from "@sdk/emoji_data";
+import { getEmojicoinMarketAddressAndTypeTags } from "@sdk/markets";
+import { getAptosClient } from "@sdk/utils";
+
+const aptosClient = getAptosClient();
+
+export type TokenBalance = {
+  amount: string;
+  asset_type: string;
+  owner_address: string;
+  metadata: {
+    symbol: string;
+    decimals: number;
+  };
+};
+
+export type FetchEmojicoinBalancesResponse = {
+  current_fungible_asset_balances: TokenBalance[];
+};
+
+export type FetchFungibleAssetsParams = {
+  ownerAddress?: string;
+  // Can be used to fetch holders of specific coin.
+  assetType?: string;
+  offset?: number;
+  limit?: number;
+};
+
+export async function fetchFungibleAssetsBalance({
+  ownerAddress,
+  assetType,
+  offset = 0,
+  limit = 100,
+}: FetchFungibleAssetsParams): Promise<FetchEmojicoinBalancesResponse> {
+  const conditions = ['{ metadata: {token_standard: {_eq: "v1"}} }', '{ amount: {_gt: "0"} }'];
+
+  if (assetType) conditions.push("{ asset_type: {_eq: $assetType} }");
+  else conditions.push('{ asset_type: {_ilike: "%coin_factory::Emojicoin%"} }');
```

Comment:
```suggestion
  else conditions.push('{ asset_type: {_ilike: "%::coin_factory::Emojicoin"} }');
```

Are you fetching both the regular `Emojicoin` *and* `EmojicoinLP` coins here? If not, you can make this more efficient and accurate by removing the end `%`, since the coin type should end there

I believe you can also make it slightly more specific with the initial `::`, too